scheduler/src/cook/quota.clj

```diff
@@ -142,22 +143,70 @@
                           count-field
                           (conj [:db/retract [:quota/user user] :quota/count count-field]))))))
 
+(defn- pool-name->type->user->quota
+  "Returns the user->quota for all quota specified for the given pool and resource type."
+  [db pool-name resource-type]
+  (let [type (resource-type->datomic-resource-type resource-type)
+        pool-name' (pool/pool-name-or-default pool-name)
+        requesting-default-pool (pool/requesting-default-pool? pool-name)
+        query '[:find ?u ?a
+                :in $ ?t ?pool-name ?requesting-default-pool
+                :where
+                [?e :quota/user ?u]
+                [?e :quota/resource ?r]
+                [?r :resource/type ?t]
+                [?r :resource/amount ?a]
+                [(cook.pool/check-pool $ ?r :resource/pool ?pool-name ?requesting-default-pool)]]]
+    (->> (d/q query db type pool-name' requesting-default-pool)
+      (into {}))))
+
+(defn- retrieve-count-quota-from-user-entities
+  "Returns the user->count-quota for all count quota specified on the user entity."
+  [db users]
+  (let [query '[:find ?u ?c
+                :in $ [?u ...]
+                :where
+                [?e :quota/user ?u]
+                [?e :quota/count ?c]]]
+    (->> (d/q query db users)
+      (into {}))))
+
+(defn- retrieve-user->count-quota
```

Comment:
Inconsistent naming with prior function?

Reply:
Renamed all the functions to `retrieve-user->*` and updated the docstrings.